bot/coze/coze.py

```diff
@@ -0,0 +1,97 @@
+# encoding:utf-8
+
+import requests
+import json
+from common import const
+from bot.bot import Bot
+from bot.session_manager import SessionManager
+from bridge.context import ContextType
+from bridge.reply import Reply, ReplyType
+from common.log import logger
+from config import conf
+from bot.coze.coze_session import CozeSession
+
+COZE_API_KEY = conf().get("coze_api_key")
+COZE_BOT_ID = conf().get("coze_bot_id")
+
+class CozeBot(Bot):
+
+    def __init__(self):
+        super().__init__()
+        self.sessions = SessionManager(CozeSession, model="coze")
+
+    def reply(self, query, context=None):
+        # acquire reply content
+        if context and context.type:
+            if context.type == ContextType.TEXT:
+                # logger.info("[COZE] query={}".format(query))
+                session_id = context["session_id"]
+                reply = None
+                if query == "#清除记忆":
+                    self.sessions.clear_session(session_id)
+                    reply = Reply(ReplyType.INFO, "记忆已清除")
+                elif query == "#清除所有":
+                    self.sessions.clear_all_session()
+                    reply = Reply(ReplyType.INFO, "所有人记忆已清除")
+                else:
+                    session = self.sessions.session_query(query, session_id)
+                    result = self.reply_text(query)
+                    total_tokens, completion_tokens, reply_content = (
+                        result["total_tokens"],
+                        result["completion_tokens"],
+                        result["content"],
+                    )
+                    logger.debug(
+                        "[COZE] new_query={}, session_id={}, reply_cont={}, completion_tokens={}".format(session.messages, session_id, reply_content, completion_tokens)
+                    )
+
+                    if total_tokens == 0:
+                        reply = Reply(ReplyType.ERROR, reply_content)
+                    else:
+                        self.sessions.session_reply(reply_content, session_id, total_tokens)
+                        reply = Reply(ReplyType.TEXT, reply_content)
+                return reply
+            elif context.type == ContextType.IMAGE_CREATE:
+                ok, retstring = self.create_img(query, 0)
+                reply = None
+                if ok:
+                    reply = Reply(ReplyType.IMAGE_URL, retstring)
+                else:
+                    reply = Reply(ReplyType.ERROR, retstring)
+                return reply
+
+    def reply_text(self, session: str, retry_count=0):
+        try:
+            # logger.info("[COZE] model={}".format(session.model))
+            url = "https://api.coze.cn/open_api/v2/chat"
```

Comment:
Is this URL still available?
Refer to:
https://www.coze.cn/docs/developer_guides/chat_v3
I think it should be https://api.coze.cn/v3/chat